contracts/strategies/ElevenStrategyForLPV1.sol

```diff
@@ -0,0 +1,206 @@
+    IPair private immutable swapPairToken0;
+    IPair private immutable swapPairToken1;
+    uint private immutable PID;
+    address private constant WAVAX = 0xB31f66AA3C1e785363F0875A1B74E27b85FD66c7;
+
+    constructor (
+        string memory _name,
+        address _depositToken,
+        address _rewardToken,
+        address _stakingContract,
+        address _vaultContract,
+        address _swapPairToken0,
+        address _swapPairToken1,
+        uint _pid,
+        uint _minTokensToReinvest,
+        uint _adminFeeBips,
+        uint _devFeeBips,
+        uint _reinvestRewardBips
+    ) {
+        name = _name;
+        depositToken = IERC20(_depositToken);
+        rewardToken = IERC20(_rewardToken);
+        stakingContract = IElevenChef(_stakingContract);
+        vaultContract = IElevenGrowthVault(_vaultContract);
+        PID = _pid;
+        devAddr = msg.sender;
+
+        swapPairToken0 = IPair(_swapPairToken0);
+        swapPairToken1 = IPair(_swapPairToken1);
+
+        setAllowances();
+        updateMinTokensToReinvest(_minTokensToReinvest);
+        updateAdminFee(_adminFeeBips);
+        updateDevFee(_devFeeBips);
+        updateReinvestReward(_reinvestRewardBips);
+        updateDepositsEnabled(true);
+
+        emit Reinvest(0, 0);
+    }
+
+    function totalDeposits() public override view returns (uint) {
+        return deployedLPBalance();
+    }
+
+    function setAllowances() public override onlyOwner {
+        depositToken.approve(address(vaultContract), type(uint256).max);
+        IERC20(address(vaultContract)).approve(address(stakingContract), type(uint256).max);
+    }
+
+    function deposit(uint amount) external override {
+        _deposit(msg.sender, amount);
+    }
+
+    function depositWithPermit(uint amount, uint deadline, uint8 v, bytes32 r, bytes32 s) external override {
+        depositToken.permit(msg.sender, address(this), amount, deadline, v, r, s);
+        _deposit(msg.sender, amount);
+    }
+
+    function depositFor(address account, uint amount) external override {
+        _deposit(account, amount);
+    }
+
+    function _deposit(address account, uint amount) private onlyAllowedDeposits {
+        require(DEPOSITS_ENABLED == true, "ElevenStrategyV1::_deposit");
+        if (MAX_TOKENS_TO_DEPOSIT_WITHOUT_REINVEST > 0) {
+            uint unclaimedRewards = checkReward();
+            if (unclaimedRewards > MAX_TOKENS_TO_DEPOSIT_WITHOUT_REINVEST) {
+                _reinvest(unclaimedRewards);
+            }
+        }
+        require(depositToken.transferFrom(msg.sender, address(this), amount), "ElevenStrategyV1::transfer failed");
+        _mint(account, getSharesForDepositTokens(amount));
+        _stakeDepositTokens(amount);
+        emit Deposit(account, amount);
+    }
+
+    function withdraw(uint amount) external override {
+        require(amount > 0, "ElevenStrategyV1::withdraw");
+        uint elevenShares = _convertSharesToElevenShares(amount);
```

Comment:
does this consider the withdrawal fee?

Reply:
No - withdraw fee is taken from vaultContract withdraw (see L119)